README.md

```diff
@@ -46,6 +46,57 @@ and are automatically released (multiple times a day, when needed) by
 Type checkers should be able to use these stub packages when installed. For more
 details, see the documentation for your type checker.
 
+### Package versioning for third-party stubs
+
+Version numbers of third-party stub packages consist of at least four parts.
+All parts of the stub version, except for the last part, correspond to the
+version of the runtime package being stubbed. For example, if the `types-foo`
+package has version `1.1.2.0`, this signifies that the package contains stubs
+for `foo==1.1.2`. In this example, the final element of the version
+indicates that this is the first revision of the stubs for `foo`. If an update
+to the stubs were pushed (but the stubs were still aiming to provide
+annotations for `foo==1.1.2`), then the version of `types-foo` would
+increment to `1.1.2.1`.
```

Comment:
What happens if `foo` at runtime has version `1.1.1`, and typeshed has `version = 1.1.*` in `stubs/foo/METADATA.toml`? Will the version for `types-foo` be `1.1.0.whatever`, or `1.1.1.whatever`?

Reply:
Hmm, looks like the version numbers can get out of sync in cases like that:
- `Deprecated` has version `2.1.14` at runtime: https://pypi.org/project/Deprecated/
- https://github.com/python/typeshed/blob/54a8fd74a785ff7916bade3699a2be4542f1b1c0/stubs/Deprecated/METADATA.toml#L1
- `types-Deprecated` is currently on version `2.1.9.3`: https://pypi.org/project/types-Deprecated/.

Reply:
Maybe something like this?

```suggestion
version of the runtime package being stubbed. For example, if the `types-foo`
package has version `1.1.2.0`, this signifies that the package contains stubs
for `foo>=1.1.2`, but which have not been tested against `foo`>=1.2`.
In this example, the final element of the version indicates that this is the
first revision of the stubs for `foo>=1.1.2`. If an update to the stubs were
pushed (but the stubs were still aiming to provide annotations for
`foo>=1.1.2`), then the version of `types-foo` would increment to `1.1.2.1`.
```

Reply:
There is some legacy there because of interaction with previous versioning. stub_uploader sees a) the specifier `"1.2.*"`, b) the fact that the last thing we uploaded was `1.2.9.3` (and at time of version swap last thing we uploaded was `1.2.9`) and just bumps.

The next time stubsabot bumps the version specifier, everything will become good. Specifier will be `1.3.*`, `Deprecated` will be `1.3.x` and we will be `1.3.0.y`.

Note the cause the complexity here is that we don't maintain multiple versions of stubs, so we do want to have each version of stub be workable for a specifier, but we can't put `*` into versions, so (assuming no legacy) we put `0` if possible